Document axes (#201)
openfisca/openfisca-doc: source/running-simulation.md

```

For this to work, the `count` values of parallel axes must be the same. An error will be raised if they are different.

### Adding axes: perpendicular axes

Sets of axes in the outer array are "perpendicular", that is, they result in independent variation. For instance, we might have:

```python
WITH_PERPENDICULAR_AXES = {
    'persons': {'Ari': {}, 'Paul': {}, 'Leila': {}, 'Javier': {}},
    'households': {
        'h1': {'children': ['Leila'], 'parents': ['Ari', 'Paul']},
        'h2': {'parents': ['Javier']}
        },
    'axes': [
        [{'count':4, 'name':'age', 'min':18, 'max':78, 'period':'2018-11'}],
        [{'count':4, 'name':'salary', 'min':0, 'max':3000, 'period':'2018-11'}]
    ]
    }
```

Note the difference in nesting: we no longer have an inner set of two axes, but two sets of one axis each. The result is more complex:

```python
>>> len(simulation.calculate('salary', '2018-11'))
64
```

Why? Because we are varying `age` and `salary` independently, and each axis results in multiplying by 4 our original population of 4, resulting in 4 times 4 times 4 individuals, which is indeed 64.

What does the data look like?

```python

> >>> np.reshape(simulation.calculate('salary', '2018-11'),(16,4))

Reviewer: ```suggestion
>>> numpy.reshape(simulation.calculate('salary', '2018-11'),(16,4))
```

array([[   0.,    0.,    0.,    0.],
       [   0.,    0.,    0.,    0.],
       [   0.,    0.,    0.,    0.],
       [   0.,    0.,    0.,    0.],
       [1000.,    0.,    0.,    0.],
       [1000.,    0.,    0.,    0.],
       [1000.,    0.,    0.,    0.],
       [1000.,    0.,    0.,    0.],
       [2000.,    0.,    0.,    0.],
       [2000.,    0.,    0.,    0.],
       [2000.,    0.,    0.,    0.],
       [2000.,    0.,    0.,    0.],
       [3000.,    0.,    0.,    0.],
       [3000.,    0.,    0.,    0.],
       [3000.,    0.,    0.,    0.],
       [3000.,    0.,    0.,    0.]], dtype=float32)

> >>> np.reshape(simulation.calculate('age', '2018-11'),(16,4))

Reviewer: ```suggestion
>>> numpy.reshape(simulation.calculate('age', '2018-11'),(16,4))
```

array([[18,  0,  0,  0],
       [38,  0,  0,  0],
       [58,  0,  0,  0],
       [78,  0,  0,  0],
       [18,  0,  0,  0],
       [38,  0,  0,  0],
       [58,  0,  0,  0],
       [78,  0,  0,  0],
       [18,  0,  0,  0],
       [38,  0,  0,  0],
       [58,  0,  0,  0],
       [78,  0,  0,  0],
       [18,  0,  0,  0],
       [38,  0,  0,  0],
       [58,  0,  0,  0],
       [78,  0,  0,  0]], dtype=int32)
```

> We can see that the `age` variable varies by increments, and the `salary` variable also varies by increments, so that **all combinations are present**: for each age increment, there exists a variant with each salary increment.

Reviewer: ```suggestion
We can see that both the `age` and the `salary` variables vary by increments, so that **all combinations are present**: for each `age` increment, there exists a variant with each `salary` increment.
```

> This allows us to use OpenFisca for `multivariate observation`: charting how two variables interact to control a third, as in [this example](https://nbviewer.jupyter.org/github/adrienpacifico/adrienpacifico.github.io/blob/master/Notebooks/plotly_openfisca_cohabitants.ipynb).

Reviewer: ```suggestion
This allows us to use OpenFisca for `multivariate observation`: charting how two variables interact to control a third, as in [this example](https://mybinder.org/v2/gh/adrienpacifico/adrienpacifico.github.io/master?filepath=Notebooks/plotly_openfisca_cohabitants.ipynb).
```
